jaas/modules/src/main/java/org/apache/karaf/jaas/modules/properties/PropertiesBackingEngine.java

```diff
@@ -157,22 +159,37 @@ private List<RolePrincipal> listRoles(String name) {
         return result;
     }
 
+    private int getFirstRoleIndex(String name) {
+        if (name.trim().startsWith(PropertiesBackingEngine.GROUP_PREFIX)) {
+            return 0;
+        }
+        return 1;
+    }
+
     @Override
     public void addRole(String username, String role) {
         String userInfos = users.get(username);
         if (userInfos != null) {
-            for (RolePrincipal rp : listRoles(username)) {
-                if (role.equals(rp.getName())) {
-                    return; 
+
+            // groups don't have password and empty should be ignored
```

Comment:
A group can be empty (no role, no group, no user).

Reply:
My original comment was a bit misleading, what I meant is:
- If a **user** info is empty, we shouldn't replace `""` by `role`, or else `role` becomes the user's password.
- If a **group** info is empty, we should replace `""` by `role`.